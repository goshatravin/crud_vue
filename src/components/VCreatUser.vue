<template>
    <div class="home" v-show="toggle">
        <div class="new">
                    <div class="user-profile___btn_close">
            <input class="close__input" type="submit" value="Close" v-on:click='closePopUp'>
        </div>
        <div class="user-profile__avatar">
            <p>Your picture</p>
            <img class="avatar__src" src="#" alt="">
        </div>
        <div class="user-profile__name">
            <p>Your name</p>
            <input class="name__input" type="text" v-model="new_user.name">
        </div>
        <div class="user-profile__position">
            <p>Your position</p>
            <input class="possition__input" type="text" v-model="new_user.position">
        </div>
        <div class="user-profile__address">
            <p>Your address</p>
            <input class="address__input" type="text" v-model="new_user.address">
        </div>
        <div class="user-profile__company">
            <p>Your company</p>
            <input class="company__input" type="text" v-model="new_user.company">
        </div>
        <div class="user-profile___btn_save">
                <input class="save__input" type="submit" value="Save" v-on:click='saveNewUser'>
        </div>
        </div>
        <div class="eddit">
            <input type="text" v-model="edit_user['data'].name">
             <input type="text" v-model="edit_user['data'].position">
              <input type="text" v-model="edit_user['data'].address">
               <input type="text" v-model="edit_user['data'].company">
               <input type="submit" v-on:click='saveEdituser' >
               {{edit_user}}
        </div>
    </div>
</template>

<script>
import { mapState } from 'vuex';

export default {
    name: 'VCreatUser',
    data() {
        return {
            new_user: {
                name: 'George',
                position: 'Ceo',
                address: 'epronovskay 21',
                company: 'BALT',
            },
        };
    },
    computed: {
        ...mapState({
            toggle: (state) => state.newUser.toggle,
            edit_user: (state) => state.newUser.editUser,
        }),
    },
    methods: {
        // Close pop-up with registration form for registration new users
        closePopUp() {
            this.$store.dispatch('TOGGLE_POP_UP', false);
            this.new_user = {};
        },
        // Save new user and close pop-up
        saveNewUser() {
            this.$store.dispatch('SAVE_NEW_USER', this.new_user);
            this.new_user = {};
            this.closePopUp();
        },
        saveEdituser() {
            this.$store.dispatch('SAVE_EDITED_USER', this.edit_user);
        },
    },
};
</script>

<style>

</style>
