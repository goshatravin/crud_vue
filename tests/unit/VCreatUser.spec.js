import { shallowMount } from '@vue/test-utils';
import VCreatUser from '@/components/VCreatUser.vue';

describe('VCreatUser', () => {
    const build = () => {
        const wrapper = shallowMount(VCreatUser);

        return {
            wrapper,
            avatar: () => wrapper.find('.avatar__src'),
            name: () => wrapper.find('.name__input'),
            position: () => wrapper.find('.possition__input'),
            address: () => wrapper.find('.address__input'),
            company: () => wrapper.find('.company__input'),
            buttonSave: () => wrapper.find('.save__input'),
            buttonClose: () => wrapper.find('.close__input'),
        };
    };

    it('Renders the component', () => {
        // arrange
        const { wrapper } = build();
        // assert
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Renders main components', () => {
        // arrange
        const { avatar, name, position, address, company, buttonClose, buttonSave } = build();
        // assert
        expect(avatar().exists()).toBe(true);
        expect(name().exists()).toBe(true);
        expect(position().exists()).toBe(true);
        expect(address().exists()).toBe(true);
        expect(company().exists()).toBe(true);
        expect(buttonClose().exists()).toBe(true);
        expect(buttonSave().exists()).toBe(true);
    });
});
