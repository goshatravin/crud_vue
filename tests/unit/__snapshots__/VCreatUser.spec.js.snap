// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VCreatUser Renders the component 1`] = `
<div class="home">
  <div class="user-profile___btn_close"><input type="submit" value="Close" class="close__input"></div>
  <div class="user-profile__avatar">
    <p>Your picture</p> <img src="#" alt="" class="avatar__src">
  </div>
  <div class="user-profile__name">
    <p>Your name</p> <input type="text" class="name__input">
  </div>
  <div class="user-profile__position">
    <p>Your position</p> <input type="text" class="possition__input">
  </div>
  <div class="user-profile__address">
    <p>Your address</p> <input type="text" class="address__input">
  </div>
  <div class="user-profile__company">
    <p>Your company</p> <input type="text" class="company__input">
  </div>
  <div class="user-profile___btn_save"><input type="submit" value="Save" class="save__input"></div>
</div>
`;
